blockchain/state/statedb.go

```diff
@@ -1099,7 +1099,7 @@ func (s *StateDB) Commit(deleteEmptyObjects bool) (root common.Hash, err error)
 	for addr, stateObject := range s.stateObjects {
 		_, isDirty := s.stateObjectsDirty[addr]
 		switch {
-		case stateObject.selfDestructed || (isDirty && deleteEmptyObjects && stateObject.empty()):
+		case (stateObject.selfDestructed && stateObject.account.Type() == account.SmartContractAccountType) || (isDirty && deleteEmptyObjects && stateObject.empty()):
```

Comment:
Why this condition? does it break a test case?
Eventhough its usage is unlikely due to EIP-6780. EOAwithCode still can do selfdestruct.

Reply:
https://github.com/ethereum/go-ethereum/blob/293a300d64be3d9a1c2cc92c26fcff4089deadcd/core/state/statedb.go#L516-L518
As far as I looked at geth, only new contract can do selfdestruct.

Reply:
I moved this condition to `SelfDestruct6780()`.